kchd/controllers/system_status.py

```diff
@@ -86,19 +85,14 @@ def astoria_good(self) -> bool:
         """Determine whether astoria services are in a good state."""
         return len(self._seen_services) == len(self._required_services)
 
-    def get_enabled_leds(self) -> int:
+    def get_enabled_leds(self) -> Tuple[bool, bool, bool]:
         """Get the number of LEDs that should be enabled."""
-        led_factors = [
+        return [
             self.kchd_running,
             self.mqtt_up,
             self.astoria_good,
         ]
-        return sum([1 for factor in led_factors if factor])
 
     def get_state(self) -> Dict[KCHLED, bool]:
         """Get the state of controlled LEDs."""
-        leds_on = self.get_enabled_leds() + 2
-        return {
-            led: leds_on >= i + 1
-            for i, led in enumerate(self.leds)
-        }
+        return dict(zip(self.leds, self.get_enabled_leds()))
```

Comment:
In my experience people who are not used to Python find dict comprehensions easier to understand.

```suggestion
        return {led: state for led, state in self.get_enabled_leds().items()}
```